tests/unit/test_git_url_watcher.py

```diff
@@ -84,45 +104,120 @@ async def test_git_url_watcher_find_new_file(
     # now add a file in the repo
     _run_cmd(
         "touch my_file.txt; git add .; git commit -m 'I added a file';",
-        cwd=git_repo_path,
+        cwd=LOCAL_PATH,
     )
     # we should have some changes here now
     change_results = await git_watcher.check_for_changes()
     # get new sha
-    git_sha = _run_cmd("git rev-parse --short HEAD", cwd=git_repo_path)
+    git_sha = _run_cmd("git rev-parse --short HEAD", cwd=LOCAL_PATH)
     assert change_results == {REPO_ID: f"{REPO_ID}:{BRANCH}:{git_sha}"}
 
     await git_watcher.cleanup()
 
 
+async def test_git_url_watcher_find_tag_on_branch_succeeds(
+    event_loop: AbstractEventLoop, git_config: dict[str, Any]
+):
+    LOCAL_PATH = git_config["main"]["watched_git_repositories"][0]["url"].replace(
+        "file://localhost", ""
+    )
+    BRANCH = git_config["main"]["watched_git_repositories"][0]["branch"]
+
+    git_watcher = git_url_watcher.GitUrlWatcher(git_config)
+    await git_watcher.init()
+    # add the a file, commit, and tag
+    VALID_TAG = "staging_z1stvalid"
```

Comment:
I am not sure faker makes a lot of sense here since I need the valid tag to specifically match a previously given and defined regex. I could concatenate the "random" parts, getting them from faker, with the "known" parts, but this seems more like weird coding for the sake of using faker. Maybe elaborate on how exactly you would use faker? I surely cant do `VALID_TAG = faker.pystr()`